web/src/components/clusters/EmbeddedClusterUpgrading.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useReducer } from "react";
+import fetch from "../../utilities/fetchWithTimeout";
+import { Utilities } from "../../utilities/utilities";
+import Loader from "@components/shared/Loader";
+
+interface Props {
+  setTerminatedState: (terminated: boolean) => void;
+}
+
+interface State {
+  seconds: number;
+  reconnectAttempts: number;
+}
+
+const EmbeddedClusterUpgrading = (props: Props) => {
+  const [state, setState] = useReducer(
+    (currentState: State, newState: Partial<State>) => ({
+      ...currentState,
+      ...newState,
+    }),
+    {
+      seconds: 1,
+      reconnectAttempts: 1,
+    }
+  );
+
+  let countdown: (seconds: number) => void;
+  let ping: () => Promise<void>;
```

Comment:
Why are we initializing these functions like this vs `const countdown = () => { .... }`?